code/modules/mob/living/carbon/alien/special/facehugger.dm

```diff
@@ -1,291 +1,271 @@
+	return mask_facehugger
+
+/**
+ * # Facehugger Item
+ *
+ * The storage item for facehuggers, used to allow them to be worn as masks.
+ *
+ * An item used to represent a facehugger while in the inventory of something.  It stores the mob inside of it,
+ * keeping it safe from death.  Upon being dropped, the item will release the mob and destroy itself.  Also worth
+ * noting, throwing this item will transfer the throw's momentum and target to the facehugger, allowing for seamless
+ * throwing.  It also handles impregnating the wearer and subsequently killing the facehugger as well.
+ */
+/obj/item/clothing/mask/facehugger_item
+	name = "alien"
+	desc = "It has some sort of a tube at the end of its tail."
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "facehugger"
+	inhand_icon_state = "facehugger"
+	w_class = WEIGHT_CLASS_TINY
+	clothing_flags = MASKINTERNALS
+	throw_range = 5
+	tint = 3
+	flags_cover = MASKCOVERSEYES | MASKCOVERSMOUTH
+	layer = MOB_LAYER
+	max_integrity = INFINITY
+	flags_1 = PREVENT_CONTENTS_EXPLOSION_1
+	var/mob/living/simple_animal/hostile/facehugger/facehugger_mob
 
-	addtimer(CALLBACK(src, .proc/GoActive), rand(MIN_ACTIVE_TIME, MAX_ACTIVE_TIME))
+/obj/item/clothing/mask/facehugger_item/Initialize(parent_mob)
+	. = ..()
+	facehugger_mob = parent_mob
 
-/obj/item/clothing/mask/facehugger/proc/Die()
-	if(stat == DEAD)
+/obj/item/clothing/mask/facehugger_item/dropped(mob/user)
+	. = ..()
+	if(loc != user)
+		facehugger_mob.forceMove(get_turf(src))
+		forceMove(facehugger_mob)
+
+/obj/item/clothing/mask/facehugger_item/throw_at(atom/target, range, speed, mob/thrower, spin=1, diagonals_first = 0, datum/callback/callback, quickstart = TRUE)
+	facehugger_mob.throw_at(target, range, speed, thrower, spin, diagonals_first, callback, quickstart)
+
+/**
+ * Proc which begins the impregnation process 
+ *
+ * Checks whether or not the facehugger is sterile.  If it isn't, it will knockout the wearer and initialize the impregnation process.
+ * Regardless of the above condition, the facehugger will go on cooldown, preventing it from trying to couple for a time.
+ * Argument:
+ * * target - the wearer of the facehugger
+ */
+/obj/item/clothing/mask/facehugger_item/proc/Attach(mob/living/target)
+	if(!facehugger_mob.sterile)
+		target.take_bodypart_damage(brute = facehugger_mob.melee_damage_upper)
+		target.Unconscious(facehugger_mob.pregnation_time)
+		addtimer(CALLBACK(src, .proc/Impregnate, target), facehugger_mob.pregnation_time)
+	COOLDOWN_START(facehugger_mob, coupling_cooldown, facehugger_mob.couple_retry_time)
+
+/**
+ * Proc which will implant the embryo into the wearer if it can.
+ *
+ * Makes sure the wearer can still be impregnated and the facehugger is still capable of it, then implants an embryo.
+ * It will also log this event.  After the impregnation is done, the facehugger will fall limp and fall off of the wearer's face, dead.
+ * Argument:
+ * * target - the wearer of the facehugger and the one to be impregnated
```

Comment:
This is very detailed, most of the docs here are, and that's really nice. I do think you should be careful about comments that are just the proc again, the goal is to describe what the proc is for/what it's purpose is, rather then the precise steps it takes.

Not like, what calls it and how, more what its use is, what it exists to do.

Not really an issue here, just noticed you were like, edging on that line, and I figured I'd let you know